File: go/internal/scoring/testdata/structural-v1/go/state_coherent_after.go
package calibration

type CoherentAfter struct{ total int }

func applyCoherent(c *CoherentAfter, delta int) int {
	if delta < 0 {
		return c.total
	}
	c.total += delta
	if c.total > 100 {
		c.total = 100
	}
	return c.total
}

func (c *CoherentAfter) Advance(delta int) int { return applyCoherent(c, delta) }
func (c *CoherentAfter) AdvanceAgain(delta int) int { return applyCoherent(c, delta) }

File: go/internal/scoring/testdata/structural-v1/java/StateCoherentAfter.java
final class StateCoherentAfter {
  private int total;

  private int advanceState(int delta) {
    if (delta < 0) return total;
    total += delta;
    if (total > 100) total = 100;
    return total;
  }

  int advance(int delta) { return advanceState(delta); }
  int advanceAgain(int delta) { return advanceState(delta); }
}
